### PR TITLE
Add top-level log correlation docs page

### PR DESCRIPTION
**Pull request by felixbarny**

### CODE REVIEW

File: docs/log-correlation.asciidoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+The easiest way to get started with that is https://github.com/elastic/java-ecs-logging[Java ECS logging] which provides layouts/encoders
+for Logback, log4j2, and log4j.
+These layouts/encoders convert a log event into an ECS-compatible JSON structure.
+Using minimal Filebeat configuration,
+you can ingest these logs into Elasticsearch.
+
+Head over to the documentation to learn more: https://github.com/elastic/java-ecs-logging[Java ECS logging documentation].
+
+If you’re using plain-text logs it's recommended to parse out specific parts of the logs,
+like the timestamp, the log level, and the message by using an {ref}/ingest.html[ingest pipeline] with a {ref}/grok-processor.html[grok processor], for example.
```
Comment (bmorelli25) [inline docs/log-correlation.asciidoc]:
```suggestion
like the timestamp, the log level, and the message. One way to do this is by using an {ref}/ingest.html[ingest pipeline] with a {ref}/grok-processor.html[grok processor].
```

This sentence was getting a little too long

File: docs/log-correlation.asciidoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+[float]
+[[log-correlation-enable]]
+=== Step 2: Enable log correlation in the agent
+
+To correlate logs from your app with transactions captured by the Elastic APM Java Agent,
```
Comment (bmorelli25) [inline docs/log-correlation.asciidoc]:
```suggestion
In order to correlate logs from your application with transactions captured by the Elastic APM Java Agent,
```

File: docs/log-correlation.asciidoc [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+:ecs-ref: https://www.elastic.co/guide/en/ecs/current
+
+* {ecs-ref}/ecs-tracing.html[`transaction.id`]
+* {ecs-ref}/ecs-tracing.html[`trace.id`]
+
+[float]
+[[log-correlation-extract-ids]]
+=== Step 3: Extract the ID fields
+
+If you are using https://github.com/elastic/java-ecs-logging[Java ECS logging], there's nothing to do in this step.
```
Comment (bmorelli25) [inline docs/log-correlation.asciidoc]: 😄

File: docs/log-correlation.asciidoc [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+=== Step 1: Ingest your application logs into Elasticsearch
+
+To get started with log correlation, your logs have to be ingested into Elasticsearch first.
+The easiest way to get started with that is https://github.com/elastic/java-ecs-logging[Java ECS logging] which provides layouts/encoders
+for Logback, log4j2, and log4j.
+These layouts/encoders convert a log event into an ECS-compatible JSON structure.
+Using minimal Filebeat configuration,
+you can ingest these logs into Elasticsearch.
+
+Head over to the documentation to learn more: https://github.com/elastic/java-ecs-logging[Java ECS logging documentation].
```
Comment (eyalkoren) [inline docs/log-correlation.asciidoc]: It's linking to the same URL as before. Just want to make sure this is intentional.

Comment (felixbarny) [reply docs/log-correlation.asciidoc]: It is intentional. There are no docs on elastic.co yet.

**Pull request merged**